compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -284,6 +268,26 @@ struct setContractionOpEncoding
       return rewriter.notifyMatchFailure(
           linalgOp, "the op has preset compilation strategy, skip SetEncoding");
     }
+    if (!linalg::isaContractionOpInterface(linalgOp)) {
+      return rewriter.notifyMatchFailure(linalgOp,
+                                         "the op is not a contraction op");
+    }
```

Comment:
Should this be part of https://github.com/openxla/iree/pull/16163?

I think we can update `getContractionOpType` to `isSupportedContractionOp` and keep the logic in this PR. It returns true for cases that it could find an enum. Otherwise it returns false.